compiler/src/iree/compiler/Codegen/Common/GPU/test/gpu_fuse_and_hoist_forall.mlir

```diff
@@ -518,3 +518,31 @@ func.func @no_fuse_multi_use(%2: tensor<128x128xf16>, %3: tensor<128x128xf16>) -
 //       CHECK:   scf.forall.in_parallel
 //       CHECK:   linalg.add
 //       CHECK:   return
+
+// -----
+
+#map = affine_map<(d0) -> (d0 * 64)>
+func.func @fuse_imperfectly_aligned_unpack(%arg0: tensor<5x31xf16>, %arg1: index) -> tensor<128xf16> {
+  %c4 = arith.constant 4 : index
+  %c128 = arith.constant 128 : index
+  %c0 = arith.constant 0 : index
+  %0 = tensor.empty() : tensor<128xf16>
+  %unpack = tensor.unpack %arg0 inner_dims_pos = [0] inner_tiles = [31] into %0 : tensor<5x31xf16> -> tensor<128xf16>
```

Comment:
When this unpack is tiled with a tile size of 64, an extract_slice must be generated on its result. This is what used to cause the infinite loop.